src/include/daos/stack_mmap.h

```diff
@@ -0,0 +1,125 @@
+/**
+ * (C) Copyright 2016-2021 Intel Corporation.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause-Patent
+ */
+/**
+ * This file is part of daos
+ *
+ * src/engine/stack_mmap.h
+ */
+
+/*
+ * Implementation of an alternate and external way to allocate a stack
+ * area for any Argobots ULT.
+ * This aims to allow for a better way to detect/protect against stack
+ * overflow situations along with automatic growth capability.
+ * Each individual stack will be mmap()'ed with MAP_GROWSDOWN causing
+ * the Kernel to reserve stack_guard_gap number of prior additional pages
+ * that will be reserved for no other mapping and prevented to be accessed.
+ * The stacks are managed as a pool, using the mmap_stack_desc_t struct
+ * being located at the bottom (upper addresses) of each stack and being
+ * linked as a list upon ULT exit for future re-use by a new ULT, based on
+ * the requested stack size.
+ * The free stacks list is drained upon a certain number of free stacks or
+ * upon a certain percentage of free stacks.
+ * There is one stacks free-list per-engine to allow lock-less management.
+ */
+
```

Comment:
Since this is exported by the engine, all functions in this file should be prefixed with dss_.
This file should also be under include/daos_srv instead of include/daos since this is specific to the engine.
Functions and defines that are only local to the engine should be under src/engine/srv_internal.h.

Reply:
But these functions are not specific to the engine, and moreover now that I don't use anymore non-XStream/XStream specific one. this why both this include and src/common/stack_mmap.c source file are under "common" dirs.